diff -ur paragraphs_a/modules/paragraphs_type_permissions/paragraphs_type_permissions.module paragraphs_b/modules/paragraphs_type_permissions/paragraphs_type_permissions.module
--- paragraphs_a/modules/paragraphs_type_permissions/paragraphs_type_permissions.module	2019-05-21 10:38:54.722154600 +0100
+++ paragraphs_b/modules/paragraphs_type_permissions/paragraphs_type_permissions.module	2019-05-22 11:55:10.410171500 +0100
@@ -56,7 +56,9 @@
 
   // If the current user has access to this type/operation, return access
   // allowed, forbidden otherwise.
-  if ($account->hasPermission('bypass paragraphs type content access') || $account->hasPermission($operation . ' paragraph content ' . $type)) {
+  if ($account->hasPermission('bypass paragraphs type content access')
+      || $account->hasPermission($operation . ' paragraph content ' . $type)
+      || $account->hasPermission($operation . ' any paragraph')) {
     $permissions[$account->id()][$type][$operation] = AccessResult::allowed()->cachePerPermissions();
   }
   else {
@@ -84,7 +86,9 @@
 
   // If the current user has access to this type/op, return access allowed,
   // forbidden otherwise.
-  if ($account->hasPermission('bypass paragraphs type content access') || $account->hasPermission($op . ' paragraph content ' . $type)) {
+  if ($account->hasPermission('bypass paragraphs type content access')
+      || $account->hasPermission($op . ' paragraph content ' . $type)
+      || $account->hasPermission($op . ' any paragraph')) {
     $permissions[$account->id()][$type][$op] = AccessResult::allowed()->cachePerPermissions();
   }
   else {
diff -ur paragraphs_a/modules/paragraphs_type_permissions/paragraphs_type_permissions.permissions.yml paragraphs_b/modules/paragraphs_type_permissions/paragraphs_type_permissions.permissions.yml
--- paragraphs_a/modules/paragraphs_type_permissions/paragraphs_type_permissions.permissions.yml	2019-05-21 10:38:54.729655800 +0100
+++ paragraphs_b/modules/paragraphs_type_permissions/paragraphs_type_permissions.permissions.yml	2019-05-22 11:53:33.878119000 +0100
@@ -1,3 +1,11 @@
+create any paragraph:
+  title: 'Create any paragraph'
+delete any paragraph:
+  title: 'Delete any paragraph'
+update any paragraph:
+  title: 'Edit any paragraph'
+view any paragraph:
+  title: 'View any paragraph'
 permission_callbacks:
   - \Drupal\paragraphs_type_permissions\ParagraphsTypePermissions::globalPermissions
   - \Drupal\paragraphs_type_permissions\ParagraphsTypePermissions::paragraphTypePermissions
\ No newline at end of file
